### ISSUE TITLE
N/A displayed for the price in Aya'k'k after the Kor Sestor attack

### ISSUE THREAD

**Issue by 857b** [endless-sky/endless-sky#2705]
After the Kor Sestor attack in the Aya'k'k system during the Wanderer campaign the prices are displayed as N/A in the map even if the player can buy some goods on Varu K'prai. (version 0.9.6)

**Comment by tehhowch** [comment]
Is this before or after revisiting the system? If after, had you landed and not yet taken off again?
If it is immediately on landing, I think it is due to the map having not redrawn (e.g. an event order issue)

**Comment by 857b** [comment]
It remain even after landing and taking off the planet after revisiting. It may be because the system is displayed in grey ( the player can't refuel in this system since the spaceport has been destroyed).
![aya k k out](https://user-images.githubusercontent.com/27144402/27831557-7def0c3a-60cb-11e7-9586-9a762fa8092a.png)

**Comment by tehhowch** [comment]
In my game (completed available Wanderer content, compiled from source), I have prices in Aya'k'k:
![image](https://user-images.githubusercontent.com/20871346/27832708-44deeafe-6095-11e7-87e7-21775762904b.png)

**Comment by 857b** [comment]
I've not yet finished the wanderer campaign so i guess the statut of the
system may change later for me.

**Comment by Bladewood** [comment]
Can confirm that Varu K'prai has prices, in two saves.
But that's strange.
```
event "wanderers: kor sestor are aimless"
	...
	planet "Varu K'prai"
		attributes wanderer oil
		landscape land/beach1
		description `This is one of the only Wanderer worlds that still has large reserves of oil, albeit deep below the ocean surface where mining is difficult. Aside from the drilling companies and refineries, the largest local employer is the local government's safety oversight department, whose inspectors are constantly visiting the drilling platforms and shipping hubs to ensure that all the regulations for preventing oil spills are being followed.`
		description `	There used to be a thriving spaceport here, but it has been razed to the ground by the Kor Sestor fleet.`
		spaceport clear
	...
```
There is nothing in the Wanderer files that undoes this change.


I've just gone to Vara K'prai.
We have a problem! (Probably!)
![screenshot 2017-07-04 13 50 32](https://user-images.githubusercontent.com/16506683/27839535-fee80b5e-60bf-11e7-8fe3-cf525a139e72.png)

**Comment by 857b** [comment]
@tehhowch seem to have the same thing. For me there is effectively no more spaceport (see my previous screen shot). Here is my save, if it can help.
[aiden 7.txt](https://github.com/endless-sky/endless-sky/files/1122843/aiden.7.txt)

**Comment by 857b** [comment]
I think I found why N/A is displayed. For the prices to be displayed the system has to be inhabited:
[MapDetailPanel.cpp L495](https://github.com/endless-sky/endless-sky/blob/master/source/MapDetailPanel.cpp#L495)
`if(hasVisited && selectedSystem->IsInhabited(player.Flagship()))
{
	//...
}
else
       price = (hasVisited ? "n/a" : "?");`
But the IsInhabited function return false if there is no planet with a spaceport in the system:
[System.cpp L491](https://github.com/endless-sky/endless-sky/blob/master/source/System.cpp#L491)
`if(!planet.IsWormhole() && planet.HasSpaceport() && planet.CanLand(ship))
     return true;`

**Comment by endless-sky** [comment]
Fixed in 7d418c7 and b8bbebc.

**Comment by 857b** [comment]
Looks good to me for Aya'k'k (non tested). But what for an inhabited planet without spaceport ? (I don't know if it can happen  in the game).

**Comment by 857b** [comment]
Fixed in [b8f2032](https://github.com/endless-sky/endless-sky/commit/b8f203255e17685dfcc22c6aaee0f50c1c2feaa8)
